components/Editor.tsx

```diff
@@ -0,0 +1,35 @@
+import MonacoEditor, { Monaco, EditorProps } from "@monaco-editor/react";
+import React, { FC, useState, createContext, useContext } from "react";
+import * as m from "monaco-editor/esm/vs/editor/editor.api";
+
+const EditorContext = createContext<m.editor.IStandaloneCodeEditor | null>(
+  null
+);
+
+const Editor: FC<EditorProps> = ({ onMount, children, ...otherProps }) => {
+  const [editor, setEditor] = useState<m.editor.IStandaloneCodeEditor | null>(
+    null
+  );
+
+  function handleMount(editor: any, monaco: Monaco) {
+    setEditor(editor);
+    onMount?.(editor, monaco);
+  }
+
+  return (
+    <EditorContext.Provider value={editor}>
```

Comment:
I created a little wrapper for the Editor so that the Editor can support children and from the children you can call useEditor to get the editor instance.

This enables support for all sort of interesting children components for the editor. For example:
```jsx
<Editor>
  <LineSelector lines={lines} onChange={setLines} />
<Editor>

<Editor>
  <ContentWidget line={5}>
    <div>some content pinned to line 5</div>
  </ContentWidget>
<Editor>
```

Reply:
love this approach. opens it up for really cool stuff in the future. great thinking :)